scripts/data/gen-nodetime/src/relayer/lib/xrelayer.ts

```diff
@@ -0,0 +1,490 @@
+		const configPath = this.homedir + "/.ts-relayer";
+		try {
+			if (!fs.existsSync(configPath)) {
+				fs.mkdirSync(configPath);
+			}
+		} catch (e) {
+			throw new Error("Could not create config folder: " + e);
+		}
+	}
+	readOrCreateConfig() {
+		this.createConfigFolder();
+		try {
+			if (fs.existsSync(this.homedir + "/.ts-relayer/config.yaml")) {
+				let configFile = fs.readFileSync(
+					this.homedir + "/.ts-relayer/config.yaml",
+					"utf8"
+				);
+				return yaml.load(configFile);
+			} else {
+				let config = {
+					mnemonic: Bip39.encode(Random.getBytes(32)).toString(),
+				};
+				let configFile = yaml.dump(config);
+				fs.writeFileSync(
+					this.homedir + "/.ts-relayer/config.yaml",
+					configFile,
+					"utf8"
+				);
+				return config;
+			}
+		} catch (e) {
+			throw new Error("Failed reading config: " + e);
+		}
+	}
+	writeConfig(config) {
+		try {
+			let configFile = yaml.dump(config);
+			fs.writeFileSync(
+				this.homedir + "/.ts-relayer/config.yaml",
+				configFile,
+				"utf8"
+			);
+		} catch (e) {
+			throw new Error("Failed writing config: " + e);
+		}
+	}
+
+	// RPC Handlers
+	public async ensureChainSetup([rpcAddr, gasPrice, addrPrefix]: [
+		string,
+		string,
+		string
+	]): Promise<EnsureChainSetupResponse> {
+		try {
+			const tmClient = await Tendermint34Client.connect(rpcAddr);
+			let status = await tmClient.status();
+			const chain = {
+				chainId: status.nodeInfo.network,
+				rpcAddr,
+				addrPrefix,
+				gasPrice,
+			};
+			if (!this.config.chains) {
+				this.config.chains = [chain];
+				return { id: chain.chainId };
+			}
+			if (
+				this.config.chains &&
+				this.config.chains.find(
+					(x) => x.chainId != chain.chainId && x.rpcAddr == chain.rpcAddr
+				)
+			) {
+				throw new Error(
+					"RPC endpoint already exists with a different chain id"
+				);
+			}
+			if (
+				this.config.chains &&
+				this.config.chains.find(
+					(x) => x.chainId == chain.chainId && x.rpcAddr == chain.rpcAddr
+				)
+			) {
+				return { id: chain.chainId };
+			} else {
+				this.config.chains.push(chain);
+				return { id: chain.chainId };
+			}
+		} catch (e) {
+			throw new Error("Could not setup chain: " + e);
+		}
+	}
+	public createPath([srcID, dstID, options]: [
+		string,
+		string,
+		ConnectOptions
+	]): Path {
+		try {
+			let path = {
+				id: srcID + "-" + dstID,
+				isLinked: false,
+				src: {
+					chainID: srcID,
+					portID: options.sourcePort,
+				},
+				dst: {
+					chainID: dstID,
+					portID: options.targetPort,
+				},
+			};
+			if (!this.config.paths) {
+				this.config.paths = [];
+			}
+			this.config.paths.push({ path, options });
+			return path;
+		} catch (e) {
+			throw new Error("Could not create path:" + e);
+		}
+	}
+	public getPath([id]: [string]): Path {
+		if (this.config.paths) {
+			let path = this.config.paths.find((x) => x.path.id == id);
+			if (path) {
+				return path.path;
+			} else {
+				throw new Error("Path does not exist");
```

Comment:
This error is duplicated by another place. If we need to use it multiple times, let's assign to a var.